packages/esm-ward-app/src/config-schema.ts

```diff
@@ -1,5 +1,120 @@
-import { type ConfigSchema } from '@openmrs/esm-framework';
+import { type PersonAddress, Type, validator, validators, type ConfigSchema } from '@openmrs/esm-framework';
+import { type BentoElementType, bentoElementTypes } from './types';
 
-export const configSchema: ConfigSchema = {};
+const defaultWardPatientCard: WardPatientCardDefinition = {
+  card: {
+    id: 'default-card',
+    header: ['bed-number', 'patient-name', 'patient-age', 'patient-address'],
+  },
+  appliedTo: null,
+};
 
-export interface ConfigObject {}
+export const defaultBentoElementConfig: BentoElementConfig = {
+  addressFields: ['cityVillage', 'country'],
+};
+
+export const builtInBentoElements: BentoElementType[] = [
+  'bed-number',
+  'patient-name',
+  'patient-age',
+  'patient-address',
+  'admission-time',
+];
+
+export const configSchema: ConfigSchema = {
+  wardPatientCards: {
+    _description: 'Configure the display of ward patient cards',
+    bentoElementDefinitions: {
+      _type: Type.Array,
+      _default: [],
+      _elements: {
+        id: {
+          _type: Type.String,
+          _description: 'The unique identifier for this custom bento element',
+        },
+        elementType: {
+          _type: Type.String,
+          _description: 'The bento element type',
+          _validators: [validators.oneOf(bentoElementTypes)],
+        },
+        config: {
+          addressFields: {
```

Comment:
I really don't like this here.  A specific type of `patientCardElement` should have a configuration that takes in an address field.  Just like other types of `patientCardElement` (like Gravity) might take in a `Concept`.  This should not be baked into a generic config schema.

Reply:
A similar change was made to the queues app. I believe it was to facilitate adding validators on the fields. I'm not sure if it's possible to have a validator work by defining specific configs per element type, but I can play around with it. 
@brandones what do you think?

Reply:
@mseaton This is the way we handle this kind of thing in registration and service queues; it's the best pattern we've figured out for this.

@chibongho You will definitely want to add a `_default` value for `addressFields` so that it is optional, since indeed it would be very bad if we required implementers to configure `addressFields` for an element of type `bed-number`. We want implementers to be able to provide config like

```json
{
  "patientCardElementDefinitions": [
    { "id": "bednum", "elementType": "bed-number" },
    { "id": "addr", "elementType": "patient-address", "config": { "addressFields": ["cityVillage"] } } 
  ]
}
```

and then we can add validators that ensure that implementers *don't* provide an element config that doesn't make sense with the element type.

@mseaton The only way I can imagine supporting what you're talking about would be to have a separate definitions array for each type:

```json
{
  "patientCardBedNumberElementDefinitions": [
    { "id": "bednum" }
  ],
  "patientCardPatientAddressElementDefinitions": [
    { "id": "addr", "addressFields": ["cityVillage"] }
  ]
}
```

So that would be our other option here.

Reply:
Given the frequency with which this pattern appears, I'm definitely open to thinking about whether we can add a schema feature that supports it better. It would be really nice if the config system had some way of knowing what element config schema should be used based on the element type.

Reply:
Quick first pass. This usually comes up in the context of arrays of elements right? Maybe for arrays we can support something like the following:

```js
someKey: {
  _type: Type.Array,
  _default: [],
  _key: {
    id: {
      _type: Type.String,
      _description: 'The config-internal identifier for this element',
    },
  }
  _elements: [{
    id: 'address',  // applies to element whose key matches this
    elementType: {
      _type: Type.String,
      _description: 'The bento element type',
      _validators: [validators.oneOf(bentoElementTypes)],
    },
    config: {
        _type: Type.Array,
        _description: 'For bentoElementType "patient-address", defining which address fields to show',
    }
  }, {
    elementType: {
      _type: Type.String,
      _description: 'The bento element type',
      _validators: [validators.oneOf(bentoElementTypes)],
    },
  }]
}
```

That could be typed like:

```ts
type ConfigElement = {
  id: string;
  elementType: string;
}

type AddressConfigElement = ConfigElement & {
  id: 'address';
  config: Array<string>;
}

someKey: Array<ConfigElement>;
```